fe1-web/src/features/linked-organizations/components/ManualInputModal.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from 'react';
+import { Text, Modal, ScrollView } from 'react-native';
+import { TouchableWithoutFeedback } from 'react-native-gesture-handler';
+import { useToast } from 'react-native-toast-notifications';
+
+import { DatePicker, Input, PoPTextButton } from 'core/components';
+import ModalHeader from 'core/components/ModalHeader';
+import { Timestamp, Hash } from 'core/objects';
+import { ModalStyles, Typography } from 'core/styles';
+import { FOUR_SECONDS } from 'resources/const';
+import STRINGS from 'resources/strings';
+
+import { Challenge } from '../objects/Challenge';
+
+interface ManualInputModalProps {
+  visible: boolean;
+  onClose: () => void;
+  isInitiatingOrganizer: boolean;
+  onScanData: (qrCode: string | null) => void;
+}
+
+const ManualInputModal: React.FC<ManualInputModalProps> = ({
+  visible,
+  onClose,
+  isInitiatingOrganizer,
+  onScanData,
+}) => {
+  const toast = useToast();
+  const [manualLaoId, setManualLaoID] = useState<string>('');
+  const [manualPublicKey, setManualPublicKey] = useState<string>('');
+  const [manualServerAddress, setManualServerAddress] = useState<string>('');
+  const [manualChallengeValue, setManualChallengeValue] = useState<string>('');
+  const [manualChallengeValidUntil, setManualChallengeValidUntil] = useState<Timestamp>(
+    Timestamp.EpochNow().addSeconds(86400),
```

Comment:
Try to avoid magic numbers in the code. For time it already becomes more readable if you write `60 * 60 * 24` instead. Even better if you define constants in a central place saying something like `DAY_IN_SECONDS = 24 * HOUR_IN_SECONDS; HOUR_IN_SECONDS = 60 * MINUTE_IN_SECONDS; MINUTE_IN_SECONDS = 60`;